internal/pkg/manifest/validate.go

```diff
@@ -1459,7 +1457,20 @@ func (q SQSQueueOrBool) validate() error {
 	if q.IsEmpty() {
 		return nil
 	}
-	return q.Advanced.validate()
+	if q.Enabled != nil {
+		return nil
+	}
+	if err := q.Advanced.validate(); err != nil {
+		return err
+	}
+	if q.isFIFO() {
```

Comment:
Do we need this conditional?

I feel like `validate` is supposed to validate the manifest fields without having to know whether the queue is FIFO or not. 

I think methods like `isFIFO` has the assumption that the manifest is valid. Therefore, I'd argue that making a call to `isFIFO` in `validate` is a little backward logically 🤔

Reply:
isFIFO simply checks if manifest has any fifo config or not? Is there is any fifo configs available then only we call validate on fifo configs. does that make sense? Or I shouldn't be checking if `fifo` field has been mentioned or not at all?

Reply:
In another word, I think `validate` should just validate every fields out there. It should call `validate` on the fifo struct even if it's empty.

Reply:
Okay I get it